contracts/Core/VoteManager.sol

```diff
@@ -42,20 +42,20 @@ contract VoteManager is Initializable, ACL, VoteStorage, StateManager, IVoteMana
         require(stakerId > 0, "Staker does not exist");
         require(commitments[stakerId].epoch != epoch, "already commited");
 
+        // slither-disable-next-line reentrancy-events,reentrancy-no-eth
+        rewardManager.givePenalties(epoch, stakerId);
         // Switch to call confirm block only when block in previous epoch has not been confirmed
         // and if previous epoch do have proposed blocks
-
-        if (!blockManager.isBlockConfirmed(epoch - 1)) {
-            blockManager.confirmPreviousEpochBlock(stakerId);
-        }
-        rewardManager.givePenalties(epoch, stakerId);
-
         uint256 thisStakerStake = stakeManager.getStake(stakerId);
         if (thisStakerStake >= parameters.minStake()) {
             commitments[stakerId].epoch = epoch;
             commitments[stakerId].commitmentHash = commitment;
             emit Committed(epoch, stakerId, commitment, block.timestamp);
         }
+
+        if (!blockManager.isBlockConfirmed(epoch - 1)) {
+            blockManager.confirmPreviousEpochBlock(stakerId);
+        }
```

Comment:
how can you        givePenalties if previous block is not yet confirmed?

Dont you need previous block confirmed to give correct penalties?

Reply:
Yeah, we need confirmed block. Will make changes

Reply:
this should have been caught in tests. create an issue about it.